src/bindings/python/flux/accounting/create_db.py

```diff
@@ -27,6 +27,84 @@ def create_db(filepath):
     except sqlite3.OperationalError as e:
         logging.error(e)
         sys.exit(1)
+import math
+import time
+
+
+LOGGER = logging.basicConfig(filename="db_creation.log", level=logging.INFO)
+
+
+def add_usage_columns_to_table(
+    conn, table_name, priority_usage_reset_period=None, priority_decay_half_life=None
+):
+    # the number of columns (or 'bins') holding usage factors is determined by
+    # the following:
+    #
+    # PriorityUsageResetPeriod / PriorityDecayHalfLife
+    #
+    # each parameter represents a number of weeks by which to hold usage
+    # factors up to the time period where jobs no longer play a factor in
+    # calculating a usage factor
+    column_name = "usage_factor_period_0"
+    if priority_decay_half_life != None and priority_usage_reset_period != None:
+        num_columns = math.ceil(
+            int(priority_usage_reset_period) / int(priority_decay_half_life)
+        )
+        for i in range(num_columns):
+            alter_stmt = (
+                "ALTER TABLE "
+                + table_name
+                + " ADD COLUMN "
+                + column_name
+                + " REAL DEFAULT 0.0"
+            )
+            conn.execute(alter_stmt)
+            conn.commit()
+            column_name = "usage_factor_period_" + str(i + 1)
+    else:
+        for i in range(4):
+            alter_stmt = (
+                "ALTER TABLE "
+                + table_name
+                + " ADD COLUMN "
+                + column_name
+                + " REAL DEFAULT 0.0"
+            )
+            conn.execute(alter_stmt)
+            conn.commit()
+            column_name = "usage_factor_period_" + str(i + 1)
+
+
+def set_half_life_period_end(conn, priority_decay_half_life=None):
+    if priority_decay_half_life != None:
+        # convert number of weeks to seconds; this will be appended
+        # to the current time to represent one 'half-life' period
+        # for the first usage bin
+        half_life_period = int(priority_decay_half_life) * 604800
+        half_life_period_end = time.time() + half_life_period
+    else:
+        half_life_period = 604800
+        half_life_period_end = time.time() + half_life_period
+
+    update_stmt = """
+        UPDATE t_half_life_period_table
+        SET end_half_life_period=?
+        WHERE cluster='cluster'
```

Comment:
So my thinking was to have one DB per cluster in a facility. Each cluster would have their own unique `half_life_period`, depending on how the DB is configured when it is first created. 

My reasoning behind the `cluster` column was just to provide some additional context on what the `end_half_life_period` timestamp represented - the end of the current half-life period for a given cluster. Although, I could see why it might be clearer to just have a one-column, one-row table that has one timestamp value:

**t_half_life_period_table**

| end_half_life_period |
| -------------------- |
| 123456789.0            |  

I hope this makes sense.

Reply:
> So my thinking was to have one DB per cluster in a facility. Each cluster would have their own unique half_life_period, depending on how the DB is configured when it is first created.

Ok. Makes sense.  Is that how Slurm is configured on LC systems currently?  One usage/accounting database per cluster?  If so, that works for me.  If not, maybe we need to ping Ryan and others to see if one DB per cluster will work for them.

Reply:
Just pinged Ryan to double check. If I remember correctly, I believe it was okay to to have one usage/accounting database per cluster, but I will confirm and get back to you as soon as I can.

Reply:
Wanted to follow up on this. One usage/accounting DB per cluster is okay to start - long term, I think the plan would be to have one DB per network.